step-framework-parent/step-framework-collections/src/main/java/step/core/collections/inmemory/InMemoryCollection.java

```diff
@@ -192,7 +196,11 @@ public void createOrUpdateCompoundIndex(Map<String, Integer> fields) {
 
 	@Override
 	public void rename(String newName) {
-		// TODO Auto-generated method stub
+		if (parentFactory != null) {
+			parentFactory.renameCollection(nane, newName);
+		} else {
+			throw new UnsupportedOperationException("renaming collection is not supported for in memory collections");

```

Comment:
Not sure if we should throw an exception in this case or simply do nothing as we don't have any parent factory